File: LLDQuestions/logger/logger/src/LogProcessor.java
public abstract class LogProcessor {
    public static int INFO = 1;
    public static int DEBUG = 2;
    public static int ERROR = 3;
    LogProcessor nextProcessor;
    public LogProcessor(LogProcessor logProcessor)
    {
        this.nextProcessor = logProcessor;
    }

    public void log(int logLevel, String message)
    {
        if(nextProcessor!=null)
            nextProcessor.log(logLevel, message);
    }
}

File: DesignPatterns/Chain of Responsibility/ChainofResponsibility/LogProcessor.cs
namespace ChainofResponsibility
{
    public abstract class LogProcessor
    {
        public static int INFO = 0;
        public static int WARNING = 1;
        public static int ERROR = 2;
        public LogProcessor _processor;
        public LogProcessor(LogProcessor processor)
        {
            _processor = processor;
        }

        public virtual void Log(int logLevel, string message)
        {
            if(_processor != null)
            {
                _processor.Log(logLevel, message);
            }
        }
    }
}
